src/xo-app/pool/tab-patches.js

```diff
@@ -1,13 +1,160 @@
+  },
+  {
+    name: _('patchUpdateButton'),
+    itemRenderer: host => (
+      <ActionButton
+        btnStyle='primary'
+        handler={installAllHostPatches}
+        handlerParam={host}
+        icon='host-patch-update'
+      />
+    )
+  }
+]
+
+@connectStore(() => {
+  const getHosts = createGetObjectsOfType('host').filter(
+    (_, props) => host => props.pool.id === host.$pool
+  )
+
+  return {
+    hosts: getHosts
+  }
+})
+export default class TabPatches extends Component {
+  constructor (props) {
+    super(props)
+    this.state.missingPatches = {}
+  }
+
+  _getHosts = createFilter(
+    () => this.props.hosts,
+    () => this.state.missingPatches,
+    [ (host, missingPatches) => missingPatches[host.id] ]
+  )
+
+  _refreshMissingPatches = async () => {
+    const missingPatches = {}
+    await Promise.all(
+      map(this.props.hosts, host => (
+        getHostMissingPatches(host).then(patches => {
+          missingPatches[host.id] = patches.length
+        })
+      ))
+    )
+    this.setState({ missingPatches })
+  }
+
+  _installAllMissingPatches = () => (
+    Promise.all(
+      map(this._getHosts(), installAllHostPatches)
+    )
+  )
+
+  componentWillMount () {
+    this._refreshMissingPatches()
+  }
+
+  componentWillReceiveProps (nextProps) {
+    forEach(nextProps.hosts, (host, push) => {
+      const { id } = host
+
+      if (this.state.missingPatches[id] !== undefined) {
+        return
+      }
+
+      this.setState({
+        missingPatches: {
+          ...this.state.missingPatches,
+          [id]: 0
+        }
+      })
+
+      getHostMissingPatches(host).then(patches => {
+        this.setState({
+          missingPatches: {
+            ...this.state.missingPatches,
+            [id]: patches.length
+          }
+        })
+      })
+    })
+  }
+
+  render () {
+    const hosts = this._getHosts()
+    const noPatches = isEmpty(hosts)
 
-export default ({
-  hosts
-}) => <Container>
-  <Row>
-    <Col className='text-xs-right'>
-      {_('comingSoon')}
-    </Col>
-  </Row>
-</Container>
+    return (
+      <Container>
+        <Row>
+          <Col className='text-xs-right'>
+            <TabButton
+              btnStyle='secondary'
+              handler={this._refreshMissingPatches}
+              icon='refresh'
+              labelId='refreshPatches'
+            />
+            <TabButton
+              btnStyle='primary'
+              disabled={noPatches}
+              handler={this._installAllMissingPatches}
+              icon='host-patch-update'
+              labelId='installPoolPatches'
+            />
+          </Col>
+        </Row>
+        <Row>
+          <Col>
+            {!noPatches
+            ? (
+              <SortedTable
+                collection={hosts}
+                columns={MISSING_PATCHES_COLUMNS}
+                userData={this.state.missingPatches}
+              />
+            ) : <p>{_('patchNothing')}</p>
```

Comment:
You'll display this if you got no hosts, not if there are no patches missing for the hosts...not very sharp.